tomee/build.go

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tomee
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/paketo-buildpacks/libpak/effect"
+	"github.com/paketo-buildpacks/libpak/sbom"
+	"github.com/paketo-buildpacks/libpak/sherpa"
+
+	"github.com/heroku/color"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libjvm"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+)
+
+type Build struct {
+	Logger      bard.Logger
+	SBOMScanner sbom.SBOMScanner
+}
+
+func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
+	result := libcnb.NewBuildResult()
+
+	m, err := libjvm.NewManifest(context.Application.Path)
+	if err != nil {
+		return libcnb.BuildResult{}, fmt.Errorf("unable to read manifest\n%w", err)
+	}
+
+	if _, ok := m.Get("Main-Class"); ok {
+		for _, entry := range context.Plan.Entries {
+			result.Unmet = append(result.Unmet, libcnb.UnmetPlanEntry{Name: entry.Name})
+		}
+		return result, nil
+	}
+
```

Comment:
There was one more thing we had to do to get the BP_JAVA_APP_SERVER working correctly.

See https://github.com/paketo-buildpacks/apache-tomcat/pull/209

Reply:
I've implemented the same change as part of the latest PR